integration/src/integration/java/pl/allegro/tech/hermes/integration/MessagesPublishingStartupValidationTest.java

```diff
@@ -0,0 +1,131 @@
+    private static final int KAFKA_PORT = 9098;
+    private static final String KAFKA_URL = "localhost:" + KAFKA_PORT;
+
+    private ZookeeperStarter zookeeper;
+    private HermesAPIOperations operations;
+    private ManagementStarter hermesManagement;
+    private KafkaStarter kafkaStarter;
+
+    @BeforeClass
+    public void setupEnvironment() throws Exception {
+        zookeeper = setupZookeeper();
+        kafkaStarter = setupKafka();
+        hermesManagement = setupHermesManagement();
+        operations = setupOperations();
+    }
+
+    @AfterClass
+    public void cleanEnvironment() throws Exception {
+        hermesManagement.stop();
+        kafkaStarter.stop();
+        zookeeper.stop();
+    }
+
+    @Test
+    public void shouldReturnCorrectReadinessStatusBasedOnKafkaReadiness() throws Exception {
+        // given
+        int frontendPort = Ports.nextAvailable();
+        String frontendUrl = "http://localhost:" + frontendPort + "/";
+
+        // when
+        Throwable exception = catchThrowable(() -> setupFrontend(frontendPort));
+
+        // then
+        assertThat(exception).isInstanceOf(IllegalStateException.class);
+        assertThat(exception.getMessage()).isEqualTo("Missing topic to validate publishing messages on startup");
+
+        // when
+        Topic topic = operations.buildTopic(TopicBuilder.topic(BROKER_MESSAGES_PUBLISHING_VALIDATION_GROUP, BROKER_MESSAGES_PUBLISHING_VALIDATION_TOPIC).build());
+        assertThat(topic).isNotNull();
+
+        // when
+        kafkaStarter.stop();
+        exception = catchThrowable(() -> setupFrontend(frontendPort));
+
+        // then
+        assertThat(exception).isInstanceOf(PublishingStartupValidationException.class);
+        assertThat(exception.getMessage()).isEqualTo("Error while validating publishing messages, last result: failed:100, success:0");
+
+        // when
+        kafkaStarter.start();
+        FrontendStarter frontend = setupFrontend(frontendPort);
+        WebTarget clientHealth = JerseyClientFactory.create().target(frontendUrl).path("status").path("health");
+
+        // then
+        assertThat(clientHealth.request().get()).hasStatus(OK);
+
+        // cleanup
+        frontend.stop();
+    }
+
+    private FrontendStarter setupFrontend(int port) throws Exception {
+        FrontendStarter frontend = new FrontendStarter(port, false);
+        frontend.overrideProperty(Configs.METRICS_ZOOKEEPER_REPORTER, false);
+        frontend.overrideProperty(Configs.MESSAGES_LOCAL_STORAGE_ENABLED, false);
+        frontend.overrideProperty(Configs.MESSAGES_LOCAL_STORAGE_DIRECTORY, Files.createTempDir().getAbsolutePath());
+        frontend.overrideProperty(Configs.ZOOKEEPER_CONNECT_STRING, ZOOKEEPER_URL);
+        frontend.overrideProperty(Configs.BROKER_PUBLISHING_STARTUP_VALIDATION_ENABLED, true);
+        frontend.overrideProperty(Configs.BROKER_PUBLISHING_STARTUP_VALIDATION_TIMEOUT_MS, 5000L);
```

Comment:
are we waiting 5 seconds in this test for kafka to be alive ? maybe we should shorten this a little bit ? :)